lib/tpm2_kdfa.c

```diff
@@ -21,8 +18,7 @@ TSS2_RC tpm2_kdfa(TPMI_ALG_HASH hash_alg, TPM2B *key, char *label,
     tpm2b_i_2.size = 4;
 
     tpm2b_bits.size = 4;
-    bits_swizzled = tpm2_util_endian_swap_32(bits);
-    *(UINT32 *) tpm2b_bits_ptr = bits_swizzled;
+    *(UINT32 *)&tpm2b_bits.buffer[0] = tpm2_util_hton_32(bits);
```

Comment:
Trap representation, makes it concise though.

Reply:
What do you mean? 
Does it need a space before the `&`?

Reply:
Or do you want me to revert the changes further above and use `tpm2b_bits_ptr` ?

Reply:
@stefanberger currently fails arm32v7 build
  CC       lib/lib_libcommon_a-tpm2_kdfa.o
../../lib/tpm2_kdfa.c: In function 'tpm2_kdfa':
../../lib/tpm2_kdfa.c:21:5: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
     *(UINT32 *)&tpm2b_bits.buffer[0] = tpm2_util_hton_32(bits);
     ^
../../lib/tpm2_kdfa.c:57:9: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
         *(UINT32 *)&tpm2b_i_2.buffer[0] = tpm2_util_hton_32(i);
         ^
cc1: all warnings being treated as errors